### PR TITLE
OCM hub integration

### PR DESCRIPTION
**Pull request by maleck13**
Should be in an operational state 
Still to do

- tidy up and go through code comments etc linting etc
- rebase
- improve walkthrough

#verification 

Go through the walkthrough doc
https://github.com/Kuadrant/multicluster-gateway-controller/pull/175/files#diff-d9a0f59f1e1d060dd48a5280234d1fa3fbc36aec23bdb679af3d2884eb84e0b9

Key Points Proven out

- Status integration from spoke and hub
- Placement integration 
- Work API integration 

Remaining Work

- [ ] Get the e2e tests passing again
- [ ] Add unit tests for placement and dns service
- [ ] Update to use Raw status response

### CODE REVIEW

File: gateway_conditions.md [hunk 1]
```diff
@@ -1,134 +0,0 @@
-# GatewayClass & Gateway Status Conditions
```
Comment (maleck13) [inline gateway_conditions.md:1]: didn't mean to delete this will re-add

File: pkg/controllers/gateway/gateway_controller.go [hunk 2]
```diff
@@ -20,48 +20,65 @@ import (
...
 type CertificateService interface {
 	EnsureCertificate(ctx context.Context, host string, owner metav1.Object) error
 	GetCertificateSecret(ctx context.Context, host string, namespace string) (*corev1.Secret, error)
 }
 
-type GatewayHelper interface {
-	GetGatewayStatuses(ctx context.Context) []gatewayv1beta1.GatewayStatus
-	GetListenerByHost(host string) *gatewayv1beta1.Listener
-	GetListenerStatusesByListenerName(ctx context.Context, listenerName string) []gatewayv1beta1.ListenerStatus
+type GatwayPlacer interface {
```
Comment (philbrookes) [inline pkg/controllers/gateway/gateway_controller.go:70]: nit: typo `gateway`

File: docs/ocm-control-plane-walkthrough.md [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+- run `make local-setup MCTC_WORKLOAD_CLUSTERS_COUNT=2`
+- setup a `.controller-config.env` with the following key values
+
+```
+    # this sets up your default managed zone
+    AWS_DNS_PUBLIC_ZONE_ID=<AWS ZONE ID>
+    ZONE_ROOT_DOMAIN=some.domain.com
+```   
+
+- setup a `.aws-credentials.env` with credentials to access route 53
```
Comment (david-martin) [inline docs/ocm-control-plane-walkthrough.md:24]:
````suggestion
- setup a `./aws-credentials.env` with credentials to access route 53

For example:

```
AWS_ACCESS_KEY_ID=<access_key_id>
AWS_SECRET_ACCESS_KEY=<secret_access_key>
AWS_REGION=eu-west-1
```

````

File: docs/ocm-control-plane-walkthrough.md [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+  namespace: multi-cluster-traffic-controller-system  
+spec:
+  gatewayClassName: mctc-gw-istio-external-instance-per-cluster
+  listeners:
+  - allowedRoutes:
+      namespaces:
+        from: All
+    name: api
+    hostname: some.domain.com
+    port: 80
```
Comment (david-martin) [inline docs/ocm-control-plane-walkthrough.md:171]:
```suggestion
    port: 443
```

File: docs/ocm-control-plane-walkthrough.md [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+metadata:
+  name: http-gateway
+  namespace: multi-cluster-traffic-controller-system 
+spec:
+  numberOfClusters: 2
+  clusterSets:
+    - gateway-clusters
+EOF
+```    
+
```
Comment (david-martin) [inline docs/ocm-control-plane-walkthrough.md:142]: All of the above cmds are fine as a stopgap to get an env working with the code changes in this PR.
Let's move it all into the local-setup soon

File: docs/ocm-control-plane-walkthrough.md [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+example
+
+clusteradm join --hub-token the_token --hub-apiserver https://127.0.0.1:49710 --wait --cluster-name gateway-1 --force-internal-endpoint-lookup
+
+clusteradm join --hub-token the_token --hub-apiserver https://127.0.0.1:49710 --wait --cluster-name gateway-2 --force-internal-endpoint-lookup
+```
+Once the gateway clusters have registered you will need to accept them in the hub back in the terminal that has the hub cluster context set 
+
+```
+clusteradm accept --clusters gateway-1
```
Comment (R-Lawton) [inline docs/ocm-control-plane-walkthrough.md:65]: when registering the spoke cluster ive named them the same as what kind names them i.e kind-mctc-workload-1 i can change that to match this

**Comment by david-martin** [comment]
closes #158 and #152

**Comment by maleck13** [comment]
https://github.com/Kuadrant/multicluster-gateway-controller/issues/210

File: pkg/dns/service.go [hunk 7]
```diff
@@ -409,7 +243,7 @@ func (s *Service) GetManagedZoneForHost(ctx context.Context, host string, t traf
 		return &managedZones.Items[0], subDomain, nil
 	}
 
-	if err := s.controlClient.List(ctx, &managedZones, client.InNamespace(s.defaultCtrlNS), client.MatchingFields{"spec.domainName": parentDomain}); err != nil {
+	if err := s.controlClient.List(ctx, &managedZones, client.InNamespace(t.GetNamespace()), client.MatchingFields{"spec.domainName": parentDomain}); err != nil {
```
Comment (mikenairn) [inline pkg/dns/service.go:246]: This is exactly the same as the check above. I'd say you can just remove it, was intended as part of this anyway https://github.com/Kuadrant/multicluster-gateway-controller/issues/202. Is there a ManagedZone still created during local-setup or is that a manual step?

Comment (mikenairn) [reply pkg/dns/service.go]: OK, so the walkthrough demo must be expecting you to create your resources in the same ns, or create a new ManagedZone?

**Pull request merged**